src/finance/PoolSteward.sol

```diff
@@ -0,0 +1,191 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {IERC20} from "solidity-utils/contracts/oz-common/interfaces/IERC20.sol";
+import {OwnableWithGuardian} from "solidity-utils/contracts/access-control/OwnableWithGuardian.sol";
+import {ICollector, CollectorUtils as CU} from "aave-helpers/src/CollectorUtils.sol";
+import {AaveV3Ethereum, AaveV3EthereumAssets} from "aave-address-book/AaveV3Ethereum.sol";
+import {AaveV2Ethereum, AaveV2EthereumAssets} from "aave-address-book/AaveV2Ethereum.sol";
+import {IPoolSteward} from "./interfaces/IPoolSteward.sol";
+
+/**
+ * @title PoolSteward
+ * @author luigy-lemon  (Karpatkey)
+ * @author efecarranza  (Tokenlogic)
+ * @notice Manages deposits, withdrawals, and asset migrations between Aave V2 and Aave V3 pools
+ */
+contract PoolSteward is OwnableWithGuardian, IPoolSteward {
```

Comment:
Perhaps better `PoolExposureSteward` or sth similar? As it manages the pool exposure. Perhaps there#sa better name, but Pool alone sounds a bit vague tbh.